pkg/controllers/provisioningv2/rke2/machinenodelookup/controller.go

```diff
@@ -113,7 +114,7 @@ func (h *handler) associateMachineWithNode(_ string, bootstrap *rkev1.RKEBootstr
 
 	nodeLabelSelector := metav1.LabelSelector{MatchLabels: map[string]string{planner.MachineUIDLabel: string(machine.GetUID())}}
 	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: labels.Set(nodeLabelSelector.MatchLabels).String()})
-	if err != nil || len(nodes.Items) == 0 || nodes.Items[0].Spec.ProviderID == "" {
+	if err != nil || len(nodes.Items) == 0 || nodes.Items[0].Spec.ProviderID == "" || !condition.Cond("Ready").IsTrue(nodes.Items[0]) {
```

Comment:
Just wondering out of curiosity, is there any reason why creating the condition here is preferred over using `NodeConditionReady`?

Reply:
This was the precedent that was set by Darren. I believe the reason is:
1. To keep `v3` separate from `v1`.
2. Related: `NodeContionReady` would be imported at `v3.NodeConditionReady`. So it would look strange to have the on a `v1` object.